### PR TITLE
feat: allow CLI tuning

### PR DESCRIPTION
**Pull request by cdaringe**
# Problem

I wanted to tune my boot configuration, but could not

# Solution

Support it!

### CODE REVIEW

File: docker/entrypoint.sh [hunk 1]
```diff
@@ -1,35 +1,42 @@
 #!/bin/sh
+set -eo
+CONF_JSON=${CONF_JSON:-/opt/fsm-data/conf.json}
+PORT=${PORT:-80}
 
 init_config() {
-    jq_cmd='.'
```
Comment (cdaringe) [inline docker/entrypoint.sh:4]: oh brother. my editor formatted this 😵‍💫

File: docker/entrypoint.sh [hunk 2]
```diff
@@ -1,35 +1,42 @@
...
 install_game
 
-cd /opt/fsm && ./factorio-server-manager --conf /opt/fsm-data/conf.json --dir /opt/factorio --port 80
-
+cd /opt/fsm
+./factorio-server-manager \
+  --conf "$CONF_JSON" \
+  --dir /opt/factorio \
+  --port "$PORT" \
+  "$@"
```
Comment (cdaringe) [inline docker/entrypoint.sh:42]: ^^ this is the most important change! allow me to append params on boot

**Comment by knoxfighter** [comment]
Hi, unfortunately I don't see any use case for this PR.
All settings can already be changed either by docker itself or by using env varaibles or the conf.json.

So which issue do you address with this PR?

**Comment by cdaringe** [comment]
> So which issue do you address with this PR?

how can i add more CLI flags to the factorio bin without this PR?

**Comment by knoxfighter** [comment]
> how can i add more CLI flags to the factorio bin without this PR?

Good question, something to look into. This PR only adds additional parameters to the server-manager itself, which does nothing, since everything can be set through env variables as well.

**Comment by Mattie112** [comment]
What settings are you looking for?

And perhaps an env / config var
FACTORIO_FLAGS (string) so you can just supply `--foo=bar --bar=foo --whatever`

**Comment by cdaringe** [comment]
I was trying to pass —verbose to diagnose a mod that was failing to load. Since learned that Factorio fails silently when it is attempting to load mods and just skips them so it didn’t actually help me :-) I was just missing a dependency.

**Comment by Mattie112** [comment]
Ah yes that is really annoying, created server added some nice (endgame) mods that also change science etc. Getting in the endgame hmmmm didn't I add a mod? And yes turned out I did not have some dependencies but it was a bit to late in the game to add it at that point.

Not sure if the `--verbose` is already included but indeed for commands like that a "custom" option would be nice. "often used" things should get a nice setting but some advanced settings yeah I don't mind.

**Comment by cdaringe** [comment]
same thing happened to me. "I think Krastorio is supposed to happen pretttttty soon here?"

launched rocket, was still waiting for krast 😄

**Pull request closed without merging**